docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,652 @@
+different executions of the realm, providing a simple and efficient way to
+manage state persistence.
+
+However, it's important to note that this practice is not a blanket
+recommendation for all Gno code. It's specifically beneficial in the context of
+realms due to their persistent characteristics. In other Gno code, such as
+packages, the use of global variables is actually discouraged and may even be
+completely disabled in the future. Instead, packages should use global
+constants, which provide a safe and reliable way to define values that don't
+change.
+
+Also, be mindful not to export your global variables. Doing so would make them
+accessible for everyone to read and write, potentially leading to unintended
+side effects. Instead, consider using getters and setters to control access to
+these variables, as shown in the following pattern:
+
+```go
+// private global variable.
+var counter int
+
+// public getter endpoint.
+func GetCounter() int {
+    return counter
+}
+
+// public setter endpoint.
+func IncCounter() {
+    counter++
+}
+```
+
+In this example, `GetCounter` and `IncCounter` are used to read and increment
+the `counter` variable, respectively. This allows you to control how the
+`counter` variable is accessed and modified, ensuring that it's used correctly
+and securely.
+
+### Embrace `panic`
+
+In Gno, we have a slightly different approach to handling errors compared to Go.
+While the famous [quote by Rob
+Pike](https://github.com/golang/go/wiki/CodeReviewComments#dont-panic) advises
+Go developers "Don't panic.", in Gno, we actually embrace `panic`.
+
+Panic in Gno is not just for critical errors or programming mistakes as it is in
+Go. Instead, it's used as a control flow mechanism to stop the execution of a
+contract when something goes wrong. This could be due to an invalid input, a
+failed precondition, or any other situation where it's not possible or desirable
+to continue executing the contract.
+
+So, while in Go, you should avoid `panic` and handle `error`s gracefully, in Gno,
+don't be afraid to use `panic` to enforce contract rules and protect the integrity
+of your contract's state. Remember, a well-placed panic can save your contract
+from a lot of trouble.
+
+When you return an `error` in Gno, it's like giving back any other piece of data.
+It tells you something went wrong, but it doesn't stop your code or undo any
+changes you made.
+
+But, when you use `panic` in Gno, it stops your code right away, says it failed,
+and doesn't save any changes you made. This is safer when you want to stop
+everything and not save wrong changes.
+
+In general, it's good to use `panic()` in realms. In reusable packages, you can
+use either `panic` or `error`, depending on what you need.
+
+```go
+import "std"
+
+func Foobar() {
+    caller := std.GetOrigCaller()
+    if caller != "g1xxxxx" {
+        panic("permission denied")
+    }
+    // ...
+}
+```
+
+- TODO: suggest MustXXX and AssertXXX flows in p/.
```

Comment:
The inline to-dos can be fixed now or later. Although some people may dislike them, I personally find them beneficial when merged in this way.

Regarding write permissions, I believe you have the highest level of permissions, possibly because it's my personal fork. However, you still have the following options:

1. Open a pull request against my pull request.
2. Utilize GitHub's "suggestion" feature, which can be committed with a single click. This is my preferred method for receiving suggestions, and it also allows your authorship to be respected.